lib/android/nit_activity.nit

```diff
@@ -0,0 +1,217 @@
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_resume((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnPause
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_pause((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnStop
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_stop((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnDestroy
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_destroy((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnSaveInstanceState
+	  (JNIEnv *env, jobject java_activity, jint nit_activity, jobject saved_state)
+	{
+		Activity_on_save_instance_state((Activity)nit_activity, saved_state);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnRestoreInstanceState
+	  (JNIEnv *env, jobject java_activity, jint nit_activity, jobject saved_state)
+	{
+		Activity_on_restore_instance_state((Activity)nit_activity, saved_state);
+	}
+`}
+
+# Wrapper to our Java `NitActivity`
+extern class NativeNitActivity in "Java" `{ nit.app.NitActivity `}
+	super NativeActivity
+end
+
+redef class Sys
+	redef fun jvm `{ return global_jvm; `}
+end
+
+redef class App
+	# Known activities
+	var activities = new Array[Activity]
+
+	# The main Java Activity of this application
+	redef fun native_activity do return activities.first.native
+
+	redef fun setup do set_global_app
+
+	# Register app in C
+	private fun set_global_app import register_activity,
+	Activity.on_create, Activity.on_start, Activity.on_pause, Activity.on_resume,
+	Activity.on_destroy, Activity.on_stop,
+	Activity.on_save_instance_state, Activity.on_restore_instance_state `{
+		App_incr_ref(recv);
+		global_app = recv;
+	`}
+
+	# Create the Nit side to this new `native` Java activity, and return it to Java
+	private fun register_activity(native: NativeNitActivity): Activity
+	do
+		native = native.new_global_ref
+		var activity = new Activity(native)
+		activities.add activity
+		return activity
+	end
+end
+
+# An Android activity
+class Activity
+	# Native Java activity
+	var native: NativeActivity
+
+	# Notification from Android, the activity is created
+	#
+	# Do your normal static set up here.
+	#
+	# If available, `save_state` contains the activity's previously frozen state.
+	fun on_create(save_state: NativeBundle) do end
+
+	# Notification from Android, the activity is being re-initialized from a `save_state`
+	fun on_restore_instance_state(save_state: NativeBundle) do end
+
+	# Notification from Android to save state
+	fun on_save_instance_state(save_state: NativeBundle) do end
+
+	# Notification from Android, your app has been paused
+	fun on_pause do end
+
+	# Notification from Android, your app has been stopped
+	fun on_stop do end
+
+	# Notification from Android, `native_activity` is being destroyed
```

Comment:
is there a distinction between *app*, *activity* and *native activity* in your comments or you are just trying to be mean?